### ISSUE TITLE
sql: unique constraint is checked two times (in sql engine and in tarantool)

### ISSUE THREAD

**Issue by Khatskevich** [tarantool/tarantool#2495]
there is a bug in tarantool sql #2289 (which is about creating primary key index on all columns, regardless to what do you ask)

For example:
```
box.sql.execute('create table tst(a, b, c, PRIMARY KEY (a,b))')
    .....(ok)
box.sql.execute('INSERT INTO tst VALUES(1,1,1)')
    .....(ok)
box.sql.execute('INSERT INTO tst VALUES(1,1,1)')
    - error: 'UNIQUE constraint failed: tst.a, tst.b'
box.sql.execute('INSERT INTO tst VALUES(1,1,2)')
    - error: 'UNIQUE constraint failed: tst.a, tst.b'
```
ok, but
```
box.sql.execute('INSERT OR REPLACE INTO tst VALUES(1,1,2)')
    .....(ok)
box.space.tst:select{}
    ---
    - - [1, 1, 1]
      - [1, 1, 2]
    ...
box.space._index:select{}
    ....
    - [519, 0, 'sqlite_autoindex_tst_1', 'tree', {'unique': true, 'sql': ''}, [[0, 'scalar'],
          [1, 'scalar'], [2, 'scalar']]]
```
The last response shows that tarantool have no uniq constraint for a,b, only for a,b,c.
That means that second `INSERT INTO tst VALUES(1,1,1)` was rejected by sql.

**Comment by kyukhin** [comment]
Bug is that `INSERT OR REPLACE` should actually do `REPLACE`, not `INSERT`

**Comment by Khatskevich** [comment]
commit 48a75cba738d999bddf795c450b4623bfb4602ab fixed explicit insertions to secondary indexes (tarantool do it while value inserted into pk), but sqlite still uses opcodes "MakeRecord" and "NoConflict" to check constrains of secondary indexes on each insert.
To delete this opcodes we should delete pk from secondary indexes first() #2255

**Comment by kyukhin** [comment]
Let's try to cleanup `sqliteInt.h/struct Index`:
* Incorporate `index_def` reference
* This will make it possible to remove:
  - zName
  - aiColumn
  - tnum
  - nKeyCol
  - nColumn
* isCovering should be always 1 for Tarantool backend

Need to teach VDBE engine to understand that Tarantool's
table indexes actually always contain whole tuple (nKeyCol ==
nColumn). This should resolve double reference issue.

**Comment by kyukhin** [comment]
Done w/ fix of #2255
